File: Behavioral/Mediator/python/mediator.py
from __future__ import print_function

class User(object):
    def __init__(self, name, mediator):
        self.name = name
        self.mediator = mediator

    def send(self, message):
        mediator.ping(self.name, message)


class Mediator(object):
    def ping(self, name, message):
        print("User: {}: {}".format(name, message))

if __name__ == "__main__":
    mediator = Mediator()

    tom = User("Tom", mediator)
    mark = User("Mark", mediator)

    tom.send("What's up man?")
    mark.send("Not much!")
    

File: Behavioral/Mediator/javascript/mediator.js
class Mediator {
  send(user, message) {
    console.log(`${user} said ${message}`);
  }
}

class Animal {
  constructor(name, mediator) {
    this.name = name;
    this.mediator = mediator;
  }

  send(message) {
    this.mediator.send(this.name, message);
  }
}

const mediator = new Mediator();

const dog = new Animal('Dog', mediator);
const cat = new Animal('Cat', mediator);

dog.send('hi cat');
cat.send('hi dog');
